### ISSUE TITLE
Disease idea: Todo experience formula shouldn't reward procrastination

### ISSUE THREAD

**Issue by mgatten** [HabitRPG/habitica#204]
Vote for this idea on Trello: https://trello.com/c/QmhxaBRE

When an item is on the Todo list for a long time it becomes more valuable - which makes total sense since you want to reward people for overcoming their procrastination and because it's harder things that tend to go longer.

But there's no downside to putting it off even longer, which kind of runs counter to the whole point of the site. It breeds the thought, "Anything worth doing tomorrow is worth even more doing next week!"

So, an idea: Disease.

The longer a Todo is left untreated, the more likely it is to become infected. It still becomes worth more and more experience to get it done, but the longer you wait the more it might also hurt you. 

Each day, every undone Todo has a random chance of adding a disease  point to a disease meter like the HP and Exp meters. The longer it has remained undone, the more likely it is to add a point. 

The disease points remain even after the Todo that caused them has been marked as done. 

Every hour you lose a random number of hitpoints between zero and your current number of disease points divided by 24. This slow rot will kill you if left untreated, and the more you let it get out of hand the more quickly fatal it will become.

To instantly cure all disease, you can buy a "Cure Disease" spell, or you can level up. The other way to cure disease points is to get stuff done: For every three Todos you complete, you heal one point of disease. Perhaps you can also heal a point by successfully completing all of your daily tasks for a day. 

Even after healing, your old undone Todos will each still have a chance of giving you more disease at the next daily rollover. 

Of course, sometimes we add a Todo that we know we can't get to anytime soon. (I have a couple that I can't possibly begin work on for at least a month because of situations beyond my control, and I'm sure others do too.) To make the disease system fair (and, actually, to make the Experience awards fair for that matter) we should be able to enter a delay date, like, "I can't start this until xx/xx/xxxx." Then it neither gains in value nor begins accumulating disease until that date. (Actually, it might be worth MORE XP prior to that date to reward getting things done earlier than you thought possible.)

**Comment by lefnire** [comment]
I love it! Ha, color those sprites green, trello this shit

**Comment by Nelmalih0** [comment]
I understand your point but some todos you put could be things that you might be able to do them right away, like i have a friend that forgot something at my house so i put it in my todos so i wouldn't forget, but he was out of town for a few days
most of my todos i put are things that i can't do today

maybe if this is a feature you toggle on and off

**Comment by mgatten** [comment]
That's exactly right. And Todos that you handle right away aren't bad for you. And Todos that take a long time should indeed be worth more, just as they are now. But you shouldn't be rewarded for procrastinating on a Todo that you could do this week but put off until next month. By adding a small amount of damage in the form of Disease for Todos that languish on the list for a long time (like weeks), you provide an impetus to get around to them without taking away the reward for getting the long term ones done.

**Comment by Nelmalih0** [comment]
idk, i still think you should be able to toggle them on and off, mostly because half of my todos are very long term

**Comment by Nelmalih0** [comment]
I did just have an idea about this though, @lefnire you mentioned that you wanted to make todos that you slack off on to long Boss Battles, what if this is how the boss battle works (instead of calling it diseases)

and just to add i don't think you should loose exp from it, that seems a little unfair

**Comment by mgatten** [comment]
Who said anything about losing exp?

**Comment by mgatten** [comment]
#Nelmalih0: By the way, at this point ALL my Todos are long term. I'm sure most people's are, since short term ones don't (by definition) stick around long. That's why I mentioned in the original post that a field should be added for "I can't start this before xx/xx/xxxx". If you can't start it before then anyway, then you shouldn't be accumulating higher and higher exp for it, and it shouldn't become diseased. It would basically act like it was created on that day. If you check it off BEFORE that day, you might get an exp bonus for finishing early, but you should also take a hit for overestimating the time (thus discouraging entering things in future dates just for the exp bonus you know you can get by finishing before then).

**Comment by StanLindsey** [comment]
Vote for this idea on Trello: https://trello.com/c/QmhxaBRE

**Comment by Pandoro** [comment]
I see the point mgatten is making and I have to say I also though about this. I planned several "milestones" of my master thesis  as todos. Simply because I want to have them in a list somewhere and I don't want to forget some of them. I know they will be sitting around for a long time and I have already secretly rejoiced about the idea that once I finish these in something like 2 months, I'll probably get a shit load of rewards! This of course makes no real sense, but I don't want to be discouraged in any way to have my todos in my todos list. So while I like the idea that they should somehow me nerfed a bit. I'm not really sure if I like this idea really. For some things I simply cannot put a deadline or desired date and that would mean I have to keep track of these tasks somewhere else. So I really don't like the part with the date

One option I could see to make this a little less worse for the user might be that this disease does not give you a hp damage, but actually just makes the actual todo worth less. This would then mean that if you just keep on waiting and waiting, the risk becomes higher and higher that the task also becomes worth less. In the other scenario where you would start to get damage from an uncleared todo, I would simple delete it and recreate it. Which kind of defeats the purpose of the whole idea. However if I only get a smaller, but still bigger reward than that of a fresh task, I will leave it in there as it is with the disease and try to complete it as soon as possible. 

So I would like the idea if you remove the date modifier and translate hp damage to reward reduction.

**Comment by Cecron** [comment]
I like the disease idea, but I also like putting things up on my todo list when I know about them. What if I could add a "start date" that set the first day I actually could start doing anything about it. Then the todo item could be hidden until the "start date" and then it could start turning red and catch the disease. I would also have it only affect my stats once per day so I don't start surfing habitrpg.com and get more procrastination...

**Comment by wc8** [comment]
In some cases, (if I understand right, Michael Linenberger proposes this/explains it better) todos that end up on your todo list for a long time represent expired ideas and should naturally fall to the bottom of your list not become more important overtime. There shouldn't be a penalty in those cases for not doing them or deleting them.

**Comment by lefnire** [comment]
Bother, I rather liked the idea of diseases :) I'm leaving this up to the
community to decide on though

**Comment by mgatten** [comment]
@Cecron: That's pretty much precisely what I was trying to describe with the "I can't start this until xx/xx/xxxx" field. Though I don't think it shoud be hidden, because then you may forget it's there and re-add it. It should just be color coded as a future item, or maybe appear in a separate column. 

@wc8: That sort of makes sense, except that many todos are put on todo lists with full knowlege that they can't be immediately acted upon. I have todo's related to five major projects, and I can only work on one project at a time. Each project is 3-4 weeks long. As a result I have some very critical todo's that I mustn't forget about but which I know I won't be addressing for a minimum of three-to-four months. They don't fall to the bottom and become less important. On the contrary, they slowly climb to the top and consume significant portions of my day.

**Comment by wc8** [comment]
@lefnire I am glad you're not giving up on it. I didn't mean to vote against it, just point out a potential wrinkle. I think in some cases it is a good idea. 
@mgatten The idea was pointed out to me by someone else, I have to read up on M.L.'s ideas more to fully get it. I know in my own case, I have both "kinds" of tasks. In some cases, a todo was a bad or rough idea and needs pruning or more thought. In other cases tasks are daunting and one needs that reminder/push to complete them.
 If a project cannot be started for several months, it would be discouraging to start being diseased because of it. I too
I like the idea of applying it only to tasks with **start dates** and/or **due dates**.

One thing to think about: what if the start date changes, or someone else completes the task, would the user take a hit for removing the date(s)? I suggest only a small penalty if any, to make the user think about it, but not be terribly discouraged, for example, after a trip to the hospital or a cancelled project.

Update: I just noticed I missed the last paragraph of the first post which includes a restriction to dated tasks, sorry. It seems disease-on-date is ok with those who have expressed concern (trello, here), the concern arises if the disease applies to all todos. I'll update the card.

**Comment by rixx** [comment]
I'm all for due dates (in fact I came here to suggest this). This way, big tasks that take a long time will still give you loads of points, if you remember to put them in early.

After crossing the due date they could first lose their value gradually until they actually get harmful. Shouldn't be even that hard to implement once the due date is established as an attribute.

**Comment by horusofoz** [comment]
To clarify, is the current form of this idea to "disease" tasks that have a Start Date that has not yet been reached? If so, wouldn't that discourage users from recording tasks they know they need to do down the track but can't as there is a time restriction?

**Comment by mgatten** [comment]
No, the idea is to NOT have any punishment for tasks whose start dates haven't been reached. Only after a start date has come and gone, and the task has continued to languish on the list, would it slowly become infected. And there would never be a penalty against XP - it's purely an HP hit taken for diseased tasks. Rewarded XP would continue to grow.

**Comment by horusofoz** [comment]
@mgatten So the function would be adding an increase modifier to the HP penalty for tasks awaiting action once the start date is reached?

**Comment by mgatten** [comment]
A random, small, increasing one. Yes.

**Comment by wc8** [comment]
I want to update the card, for clarity; does the following accurately summarize the idea?

**Disease:**

**Vulnerable todos: any todo past its start date**: Every undone Todo which ought to have been started before today has a random chance of adding a disease point to a disease meter (like the HP & XP meters). The chance increases the longer the todo remains undone.

**Effects:** Disease points remain even after the Todo that caused them has been marked _done_. 
Every hour, you lose a random number of hit-points between 0 and the current number of Disease Points divided by 24. 
If left untreated, can result in death. The chance of death increases with time.
Even after healing, old **undone** Todos will each still have a chance of giving you more disease at the next daily rollover. Completed todos, once fully cured no longer cause disease.

**To cure disease:** level up, buy "cure disease potion", or heal 1 disease point for every 3 completed todos, completing all daily tasks for a day also heals a point. 

**Immune tasks:** Tasks with a start date after today. These tasks would also not gain value. They would also be worth more XP if completed before the start date.

**Comment by wc8** [comment]
Hm... if start date and due date are both implemented: perhaps a todo should not become diseased until after its due date or be less vulnerable to it?

**Comment by mgatten** [comment]
That description sounds perfect, including the added bit about the due date (not the start date) being the trigger.

I might suggest removing "The chance of death increases with time."  That's true, but only as a result of everything else described. Having that sentence in there might confuse some into thinking there's a random chance of instant death whereas the whole thing is meant to mimic a wasting disease. 

In order to easily tell which todos are causing your disease, it might be helpful to add a little germ icon to them or otherwise indicate the diseased status.

**Comment by wc8** [comment]
Ok, I've updated the card, incorporating your suggestions. Let me know if I missed anything.
I am much in favor of this now that I fully understand it.

Here is a question: Would it be good to have a **small** HP loss on changing the start date or due date, similar to the delete (red) daily penalty? 
There are instances in which a task suddenly **must** to be postponed, so the date(s) need(s) to be changeable, but date changes could be used as a means of procrastination.

**Comment by mgatten** [comment]
That's an excellent idea!

**Comment by antistic** [comment]
I quite like this idea too! Although perhaps "poison" might be a more RPG-like name to call it?

**Comment by mgatten** [comment]
I think disease is better. Ever since the original Dungeons & Dragons there have been Cure Disease spells, so it's RPG-appropriate. And the idea is that our Todos are like wounds that need to be tended. An untended wound will fester and eventually cause disease. Poison, on the other hand, is inherently imposed upon us by somebody else and less likely to be curable. But the name is really unimportant. The main idea is to find a way to discourage procrastination without taking away the reward for getting big things done that took a long time to get to.

**Comment by wc8** [comment]
Another possibility: introduce disease after a certain level, maybe level 4. 
The idea would be to allow users to gain a little confidence first, and introduce a new concept slowly. 
However, since diseases only come into effect after a task's start/due date it might not matter. Maybe it would introduce confusion/frustration if a user set a start date and then discovered they had to pay up to delay the task.

**Comment by Pandoro** [comment]
Personally I'm still not very happy about the idea. I can understand that for many people it works very well and for certain tasks I think it is very useful, but to me the penalty sounds to harsh. If I get it right than at some point I might get a disease point and each day I will loose 1hp/disease point. In order to cure it I will need to buy a potion or I will need to do 3 todos. 

For me, todos are things I have to do at some point, meaning doing 3 of them takes a long time, something like a week. Many todos I don't want to write down in here, as they are so plain that I don't see the added bonus in my life to add them in here. This then means I will be loosing a lot of hp, simply because I had some task which wasn't so important to me yet, but for which I did want some kind of reminder. Now I have to buy a hp potion, which for me is a LOT of gold at my current "level of playing". Or I would have to directly buy some cure at an unknown cost. So what that would mean, I'd be seriously cutting back my rewards, simply because I missed some todo. Now for some people this might be a wanted experience, but for me it certainly is not. My other option would be to enter every single habit, daily and todo I could come up with, simply to get more gold and exp so that I will survive this in any way. This however, sounds really distracting for me as I would keep thinking "can I somehow make gold and exp from this!?" all day long :-/ 

Because of that I would vote for having todos with and without due dates and start dates For some todos this might be a great to have! But please give the user also the option to opt-out and leave the dates blank for todos. Meaning that they will not be affected by disease if they know this will be some todo which will not be done any time soon. 

To make it more fair, an idea would be to let todos with start and due data grow faster in reward. This would motivate users to set these dates and thus motivate them to also work on the todos, but it would not force everyone to use the system in the same way. After all it should be adding to your productivity and I think this will not work for everyone!

**Comment by wc8** [comment]
If users have **dated tasks** that need to be not affected by disease, the feature could be limited either by 
- **High Priority**, if implemented
- **per task opt-in, that is, enable for specific tasks.

@pandoro To be clear, do you have todos which **need a date assigned** which need to be not affected by disease? The idea here is ONLY dated tasks would be affected, the user would **not** have to add a date to tasks. I better update the card for clarity.

If I understand, what @pandoro said (emphasis added) in this paragraph, is already the idea, 

> Because of that I would vote for having **todos with and without due dates and start dates** For some todos this might be a great to have! But please give the user also the option to opt-out and **leave the dates blank for todos**. Meaning that they will not be affected by disease if they know this will be some todo which will not be done any time soon.

**Comment by StanLindsey** [comment]
One of the main problems with this feature for me is the random nature of it. Users need to feel that they are in control/have player agency and understand why things are happening and the correlation between their actions and results.

The differences in how people use their to-do list make this a very hard feature to balance. That said there is some great potential.

**Comment by Pandoro** [comment]
Thank you for the clarification. I'm glad to really see this is not a mandatory thing, but rather something only for todos with dates. 

However I do think that @SlappyBag has a very good point. Let's for example consider my current todo "edit 100 images". I know cannot do it the next two weeks, but then I want to have it done within one week. Now I'm working hard on this during the one week. Every evening I'm doing about 15 images, but after 3 evenings, I get a disease point on this task. Now is this a desirable effect? Even though I am working I am getting punished. 

Let's skip the discuss whether punishment or reward is the better intensive. Would it not be better to be punished only if one actually fais to complete something one had a period of time to complete? If I failed, I would indeed like to see some penalty to show the consequences of my bad behaviour. But not for good behaviour in progress ;)

**Comment by wc8** [comment]
@pandoro the idea is for only tasks where a deadline has been missed to be vulnerable to the random disease point. In other words, penalty only on failure.  

I'm not sure what I think about "random": it's interesting that the task might not give me a penalty, so I might get a reprieve or it might affect me. 

Updated portion of task card:

> _**Affected todos:_\* **only dated** todos with a missed deadline, either:
> - todo past its due date
> - todo past its start date, if no due date is applied
> 
> **Immune tasks:** all undated tasks, 
> todos with a start date after today or no start/due date. 
> These dated tasks would also not gain value with passing time. 
> Completing a task before its start/due date would be worth more XP.

**Comment by wc8** [comment]
**Correction:** Immune tasks: all undated tasks, tasks with a **start/due** date after today are immune to disease.

**Comment by Pandoro** [comment]
Aaah okay! Sorry for the mess and confusion here ^^; Now it is clear to me. Well for that I would say: I like it!

To be honest I am somehow missing an example for a todo with a start date but no due date though. But if the main "target" of this penalty is really missing the final due date, then I would say it makes total sense to give a negative effect if the date is missed.

**Comment by wc8** [comment]
The questions are helpful. I think I was able make the card clearer so subsequent users will understand the idea better.

**Comment by wc8** [comment]
It might not make sense to disease a task with a start date but no due date.

**Comment by ghost** [comment]
@wc8 anything else? This sounds pretty good as is

**Comment by wc8** [comment]
I left the start date just in case it's helpful—maybe a user wants to reserve due dates for todos with actual deadlines, but give a sense of urgency to tasks by committing to start on a certain date. 

I updated the title also to specify missed deadlines, as it may help reduce misapprehension that it is meant to apply to all todos.

**Comment by Cecron** [comment]
I would be using the start date, but not necessarily the done date and I would like the disease feature. 
The goal for me would be to get a habit where I do stuff within a reasonable amount of time without the need to set up hard dates for completion. The "reasonable amount" would nicely be accented by the randomness of the incubation time before the disease hits me. I would expect it to be a couple of days, with a larger chance of disease for each coming day.

I would also like the red todos, those that I have neglected for a really long time, to have a chance of catching the disease even though they have no start nor due date (as described in the original proposal).

**Comment by wc8** [comment]
The problem is things like, "label kitchen drawers". I want to remember it, and if/when possible, get around to it, but I don't want to get diseased for it. 
What if todos with only a start date should only become vulnerable to disease when they turn red?
That way, users wanting all red tasks to be vulnerable to disease could enter a start date from the beginning.
Users who have tasks they don't want to ever be diseased could simply not enter a start date.

**Comment by horusofoz** [comment]
@wc8 Could your scenario be resolved via a pending flag such as that proposed [here](https://github.com/lefnire/habitrpg/issues/243)?

**Comment by wc8** [comment]
@horusofoz The theoretical task _label drawers_ isn't on hold, and I'm fine with it gaining value as a motivation to get around to it. However, it's not important enough for me to deserve to suffer a disease penalty for not doing it. In fact, if I did labels before _pay bills_ I would possibly/likely deserve to be diseased for not paying my bills on time.

On the other hand, in @Cecron's case, the preference would be for the theoretical _label drawers_ to be vulnerable disease once it turned red. I was trying to find a way to enable both methods, but I haven't got it yet. 

I wonder if priority settings could help?

On a side note, I wish todos didn't turn red so soon (if it's screaming red in less than a fortnight, what happens after a month?) perhaps start them blue > green > yellow > orange > red > intense red.

**Comment by horusofoz** [comment]
@wc8 I think priority as a trigger for disease is a good idea. Should be beasy to implement and easy for a user to understand.

**Comment by wc8** [comment]
Ok, latest version of Affected Todos section reads:

> Affected todos: only dated todos with a missed deadline, either:
> - todo past its due date
> - todo past its start date, if no due date is applied
> - todo with a high priority

If this conflict's with anyone's needs, please reply. Full card visible [here](https://trello.com/c/QmhxaBRE).

**Comment by Cecron** [comment]
I don't really see the use case for the "todo with a high priority". Wasn't the objection something like, assuming no dates have been set: if some _other_ task had _higher_ priority and I worked on that other higher priority task, not necessarily meaning I ticked it off, I don't want the _lesser_ prioritized tasks to catch a disease? Also, while working on the _higher_ priority task, and not ticking it of, that _higher_ priority task would be affected by disease.

However, this would still suite my needs if the settings could give me a default start date of tomorrow (or better yet today + n days). The reason being that I want all of my tasks to be affected by default. I want to be forced to do something extra when adding tasks that should not be affected by disease.
